app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java

```diff
@@ -155,57 +158,29 @@ public static void playOnBackgroundPlayer(final Context context,
         intent.putExtra(Player.PLAYER_TYPE, MainPlayer.PlayerType.AUDIO.ordinal());
         ContextCompat.startForegroundService(context, intent);
     }
-
-    public static void enqueueOnVideoPlayer(final Context context, final PlayQueue queue,
-                                            final boolean resumePlayback) {
-        enqueueOnVideoPlayer(context, queue, false, resumePlayback);
-    }
-
-    public static void enqueueOnVideoPlayer(final Context context, final PlayQueue queue,
-                                            final boolean selectOnAppend,
-                                            final boolean resumePlayback) {
-
+    /* ENQUEUE */
+    public static void enqueueOnPlayer(final Context context, final PlayQueue queue) {
         Toast.makeText(context, R.string.enqueued, Toast.LENGTH_SHORT).show();
-        final Intent intent = getPlayerEnqueueIntent(
-                context, MainPlayer.class, queue, selectOnAppend, resumePlayback);
-
-        intent.putExtra(Player.PLAYER_TYPE, MainPlayer.PlayerType.VIDEO.ordinal());
-        ContextCompat.startForegroundService(context, intent);
-    }
+        final Intent intent = getPlayerEnqueueIntent(context, MainPlayer.class, queue);
 
-    public static void enqueueOnPopupPlayer(final Context context, final PlayQueue queue,
-                                            final boolean resumePlayback) {
-        enqueueOnPopupPlayer(context, queue, false, resumePlayback);
-    }
-
-    public static void enqueueOnPopupPlayer(final Context context, final PlayQueue queue,
-                                            final boolean selectOnAppend,
-                                            final boolean resumePlayback) {
-        if (!PermissionHelper.isPopupEnabled(context)) {
-            PermissionHelper.showPopupEnablementToast(context);
-            return;
+        int playerType = MainPlayer.PlayerType.AUDIO.ordinal();
+        if (PlayerHolder.getInstance().getType() != null) {
+            playerType = PlayerHolder.getInstance().getType().ordinal();
         }
-
-        Toast.makeText(context, R.string.enqueued, Toast.LENGTH_SHORT).show();
-        final Intent intent = getPlayerEnqueueIntent(
-                context, MainPlayer.class, queue, selectOnAppend, resumePlayback);
-        intent.putExtra(Player.PLAYER_TYPE, MainPlayer.PlayerType.POPUP.ordinal());
+        intent.putExtra(Player.PLAYER_TYPE, playerType);
         ContextCompat.startForegroundService(context, intent);
     }
+    /* NEXT */
```

Comment:
... and add a helper function here
```suggestion

    /* ENQUEUE */
    public static void enqueueOnPlayer(final Context context, final PlayQueue queue) {
        PlayerType playerType = PlayerHolder.getInstance().getType();
        if (playerType == null) {
            Log.e(TAG, "Enqueueing when no player is open, defaulting to background");
            playerType = MainPlayer.PlayerType.AUDIO;
        }
        
        enqueueOnPlayer(context, queue, playerType);
    }

    /* ENQUEUE NEXT */
```
This function should be used anywhere except for the background and popup long-press actions (also on playlists, history, ...), where the other one should be used instead.

Reply:
This implies that the helper function will only be used in the StreamDialogEntry Class.
see commit ca17b02e8642af85f5043b738b37b12915d3580d